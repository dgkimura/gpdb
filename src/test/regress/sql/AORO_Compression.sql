-- Given that we built with and have zlib compression available
-- Test basic create table for AO/RO table succeeds for zlib compression
create table a_aoro_table_with_zlib_compression(col text) WITH (APPENDONLY=true, COMPRESSTYPE=zlib, compresslevel=1);
select pg_size_pretty(pg_relation_size('a_aoro_table_with_zlib_compression')),
       get_ao_compression_ratio('a_aoro_table_with_zlib_compression');
insert into a_aoro_table_with_zlib_compression values('ksjdhfksdhfksdhfksjhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhh');
select pg_size_pretty(pg_relation_size('a_aoro_table_with_zlib_compression')),
       get_ao_compression_ratio('a_aoro_table_with_zlib_compression');


-- Given that we built with and have zstd compression available
-- Test basic create table for AO/RO table succeeds for zstd compression
create table a_aoro_table_with_zstd_compression(col text) WITH (APPENDONLY=true, COMPRESSTYPE=zstd, compresslevel=1);
select pg_size_pretty(pg_relation_size('a_aoro_table_with_zstd_compression')),
       get_ao_compression_ratio('a_aoro_table_with_zstd_compression');
insert into a_aoro_table_with_zstd_compression values('ksjdhfksdhfksdhfksjhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhh');
select pg_size_pretty(pg_relation_size('a_aoro_table_with_zstd_compression')),
       get_ao_compression_ratio('a_aoro_table_with_zstd_compression');


-- Given that we built with and have quicklz compression available
-- Test basic create table for AO/RO table succeeds for quicklz compression
create table a_aoro_table_with_quicklz_compression(col text) WITH (APPENDONLY=true, COMPRESSTYPE=quicklz, compresslevel=1);
select pg_size_pretty(pg_relation_size('a_aoro_table_with_quicklz_compression')),
       get_ao_compression_ratio('a_aoro_table_with_quicklz_compression');
insert into a_aoro_table_with_quicklz_compression values('ksjdhfksdhfksdhfksjhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhh');
select pg_size_pretty(pg_relation_size('a_aoro_table_with_quicklz_compression')),
       get_ao_compression_ratio('a_aoro_table_with_quicklz_compression');


-- Test basic create table for AO/RO table fails for rle compression. rle is only supported for columnar tables.
create table a_aoro_table_with_rle_type_compression(col text) WITH (APPENDONLY=true, COMPRESSTYPE=rle_type, compresslevel=1);
