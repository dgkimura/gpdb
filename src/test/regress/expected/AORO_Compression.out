-- Given that we built with and have zlib compression available
-- Test basic create table for AO/RO table succeeds for zlib compression
create table a_aoro_table_with_zlib_compression(col text) WITH (APPENDONLY=true, COMPRESSTYPE=zlib, compresslevel=1);
NOTICE:  Table doesn't have 'DISTRIBUTED BY' clause -- Using column named 'col' as the Greenplum Database data distribution key for this table.
HINT:  The 'DISTRIBUTED BY' clause determines the distribution of data. Make sure column(s) chosen are the optimal data distribution key to minimize skew.
select pg_size_pretty(pg_relation_size('a_aoro_table_with_zlib_compression')),
       get_ao_compression_ratio('a_aoro_table_with_zlib_compression');
 pg_size_pretty | get_ao_compression_ratio 
----------------+--------------------------
 0 bytes        |                        1
(1 row)

insert into a_aoro_table_with_zlib_compression values('ksjdhfksdhfksdhfksjhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhh');
select pg_size_pretty(pg_relation_size('a_aoro_table_with_zlib_compression')),
       get_ao_compression_ratio('a_aoro_table_with_zlib_compression');
 pg_size_pretty | get_ao_compression_ratio 
----------------+--------------------------
 56 bytes       |                     1.57
(1 row)

-- Given that we built with and have zstd compression available
-- Test basic create table for AO/RO table succeeds for zstd compression
create table a_aoro_table_with_zstd_compression(col text) WITH (APPENDONLY=true, COMPRESSTYPE=zstd, compresslevel=1);
NOTICE:  Table doesn't have 'DISTRIBUTED BY' clause -- Using column named 'col' as the Greenplum Database data distribution key for this table.
HINT:  The 'DISTRIBUTED BY' clause determines the distribution of data. Make sure column(s) chosen are the optimal data distribution key to minimize skew.
select pg_size_pretty(pg_relation_size('a_aoro_table_with_zstd_compression')),
       get_ao_compression_ratio('a_aoro_table_with_zstd_compression');
 pg_size_pretty | get_ao_compression_ratio 
----------------+--------------------------
 0 bytes        |                        1
(1 row)

insert into a_aoro_table_with_zstd_compression values('ksjdhfksdhfksdhfksjhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhh');
select pg_size_pretty(pg_relation_size('a_aoro_table_with_zstd_compression')),
       get_ao_compression_ratio('a_aoro_table_with_zstd_compression');
 pg_size_pretty | get_ao_compression_ratio 
----------------+--------------------------
 64 bytes       |                     1.38
(1 row)

-- Given that we built with and have quicklz compression available
-- Test basic create table for AO/RO table succeeds for quicklz compression
create table a_aoro_table_with_quicklz_compression(col text) WITH (APPENDONLY=true, COMPRESSTYPE=quicklz, compresslevel=1);
NOTICE:  Table doesn't have 'DISTRIBUTED BY' clause -- Using column named 'col' as the Greenplum Database data distribution key for this table.
HINT:  The 'DISTRIBUTED BY' clause determines the distribution of data. Make sure column(s) chosen are the optimal data distribution key to minimize skew.
select pg_size_pretty(pg_relation_size('a_aoro_table_with_quicklz_compression')),
       get_ao_compression_ratio('a_aoro_table_with_quicklz_compression');
 pg_size_pretty | get_ao_compression_ratio 
----------------+--------------------------
 0 bytes        |                        1
(1 row)

insert into a_aoro_table_with_quicklz_compression values('ksjdhfksdhfksdhfksjhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhh');
select pg_size_pretty(pg_relation_size('a_aoro_table_with_quicklz_compression')),
       get_ao_compression_ratio('a_aoro_table_with_quicklz_compression');
 pg_size_pretty | get_ao_compression_ratio 
----------------+--------------------------
 64 bytes       |                     1.38
(1 row)

-- Test basic create table for AO/RO table fails for rle compression. rle is only supported for columnar tables.
create table a_aoro_table_with_rle_type_compression(col text) WITH (APPENDONLY=true, COMPRESSTYPE=rle_type, compresslevel=1);
NOTICE:  Table doesn't have 'DISTRIBUTED BY' clause -- Using column named 'col' as the Greenplum Database data distribution key for this table.
HINT:  The 'DISTRIBUTED BY' clause determines the distribution of data. Make sure column(s) chosen are the optimal data distribution key to minimize skew.
ERROR:  rle_type cannot be used with Append Only relations row orientation
